### DISCUSSION TITLE
Flux boostrap with an OCI source

### DISCUSSION THREAD

**Discussion by SamStenton** [fluxcd/flux2#3671] [Proposals] [answered]
In Air-Gapped environments it would be beneficial to not need to rely on the need of an external git server for bootstrap and state storage.

We create our Flux repositories in a non-air-gapped environment and then `flux build` them into an OCI source stored in a registry. Within the Air-Gapped environment we then just want to deploy a particular OCI source version to the cluster. 

Currently this requires the setup of a git repository and creation of a Kustomization pointing to the OCI source. Any updates only happen via a change to the OCI source. 

Is there a future Flux out there that would allow for clusters to be bootstrapped using an OCI Source only with state either stored in blob/s3? Similar to how the `gitops beta run` command structures things.

**Comment by stefanprodan** [comment]
I’ve been experimenting with Git-less Flux bootstrap in these two projects:
- https://github.com/stefanprodan/flux-local-dev
- https://github.com/stefanprodan/flux-aio

The goal is to find the best workflow for bootstrap without Git or Buckets, just OCI.

**Comment by attilaolah** [comment]
I think a reasonable alternative would be running the `flux install --export > flux/flux-system/gotk-components.yaml` command, then manually generate the `gotk-sync.yaml` file locally. Applying these would do most of the heavy-lifting I assume, except some secret generation. Those secrets could still be generated or injected manually, but the OCI repo might not even need secrets (it might be an in-cluster repo accessible only locally).

**Comment by stefanprodan** [comment, marked as answer]
Bootstrap with OCI is now supported by [Flux Operator](https://github.com/controlplaneio-fluxcd/flux-operator)

Docs:

- [Cluster Sync from a Container Registry](https://fluxcd.control-plane.io/operator/flux-sync/#sync-from-a-container-registry)
- [Migrate from Git bootstrap](https://fluxcd.control-plane.io/operator/flux-bootstrap-migration/)
